python/dgl/heterograph.py

```diff
@@ -479,16 +530,75 @@ def get_ntype_id(self, ntype):
         -------
         int
         """
+        if self.is_unibipartite:
+            # Only check 'SRC/' and 'DST/' prefix when is_unibipartite graph is True.
+            if ntype.startswith('SRC/'):
+                return self.get_srctype_id(ntype[4:])
+            elif ntype.startswith('DST/'):
+                return self.get_dsttype_id(ntype[4:])
+            # If there is no prefix, fallback to normal lookup.
+
+        # Lookup both SRC and DST
         if ntype is None:
-            if self._graph.number_of_ntypes() != 1:
+            if self.is_unibipartite or len(self._srctypes_invmap) != 1:
                 raise DGLError('Node type name must be specified if there are more than one '
                                'node types.')
             return 0
-        ntid = self._ntypes_invmap.get(ntype, None)
+        ntid = self._srctypes_invmap.get(ntype, self._dsttypes_invmap.get(ntype, None))
         if ntid is None:
             raise DGLError('Node type "{}" does not exist.'.format(ntype))
         return ntid
 
+    def get_srctype_id(self, ntype):
+        """Return the id of the given SRC node type.
+
+        ntype can also be None. If so, there should be only one node type in the
+        SRC category. Callable even when the self graph is not uni-bipartite.
+
+        Parameters
+        ----------
+        ntype : str
+            Node type
+
+        Returns
+        -------
+        int
+        """
+        if ntype is None:
+            if len(self._srctypes_invmap) != 1:
+                raise DGLError('SRC node type name must be specified if there are more than one '
+                               'SRC node types.')
+            return 0
+        ntid = self._srctypes_invmap.get(ntype, None)
+        if ntid is None:
+            raise DGLError('SRC node type "{}" does not exist.'.format(ntype))
+        return ntid
```

Comment:
Because we sort the node types in `srctypes`, `get_srctype_id(ntype)` may not necessarily return the index of `ntype` in `srctypes`, right?  Would it cause problems?

Reply:
Unlike `ntypes`,  `srctypes` and `dsttypes` cannot be used as a mapping from type id to type name. Thus, their results are technically set, but I still return list to keep it similar to `ntypes`. The list is sorted to make the behavior deterministic. Since the type id is retrieved from `_srctypes_invmap`, it won't cause trouble.

Reply:
OK.  I think users need to be aware of that though, since `get_srctype_id` would behave very differently from `get_ntype_id`.  Could we make them consistent?